lib/thesis/notifications.ex

```diff
@@ -0,0 +1,163 @@
+defmodule Thesis.Notifications do
+  @moduledoc """
+  Builds a map of Notifications that is passed to the client side. The key should be a
+  representation of the category/type of notification (this is used to determine where and how it's
+  displayed). The value is a list of notification strings.
+
+  Example of final map:
+  %{
+    "page-settings": [],
+    "add-page": [],
+    "import-export-restore": [],
+    "other": []
+  }
+  """
+
+  alias Thesis.Backup
+  import Thesis.Config
+  import Thesis.Utilities
+  import Ecto.Query, only: [limit: 2]
+
+  @allowed_notification_keys [
+    "page-settings",
+    "add-page",
+    "import-export-restore",
+    "other",
+    :page_settings,
+    :add_page,
+    :import_export_restore,
+    :other
+  ]
```

Comment:
Allowed keys for the notifications. We can extend this later when we add more Thesis features and add notifications to those features.